libs/ledger-live-common/src/deposit/helper.ts

```diff
@@ -0,0 +1,48 @@
+import { CryptoCurrency, TokenCurrency } from "@ledgerhq/types-cryptoassets";
+import { currenciesByMarketcap, findCryptoCurrencyById, findTokenById } from "../currencies";
+import { MappedAsset, GroupedCurrency } from "./type";
+
+const groupedCurrenciesByProvider = async (data: MappedAsset[]) => {
+  const groupedCurrencies = new Map<string, GroupedCurrency>();
+
+  for (const currency of data) {
+    let groupedCurrency = groupedCurrencies.get(currency.providerId);
+    if (!groupedCurrency) {
+      groupedCurrency = {
+        providerId: currency.providerId,
+        currenciesByNetwork: [],
+      };
+      groupedCurrencies.set(currency.providerId, groupedCurrency);
+    }
+    groupedCurrency.currenciesByNetwork.push({
+      ...currency,
+      ledgerCurrency: (findCryptoCurrencyById(currency.ledgerId) ||
+        findTokenById(currency.ledgerId)) as CryptoCurrency | TokenCurrency,
+    });
+  }
+
+  const mapValues = Array.from(groupedCurrencies.entries()).map(async ([_, value]) => {
+    const currencies = value.currenciesByNetwork.map(
+      e => e.ledgerCurrency as CryptoCurrency | TokenCurrency,
+    );
+    const res = await currenciesByMarketcap(currencies);
```

Comment:
So, there is a HTTP call for every item of groupedCurrencies, that can be in the thousand ?
Could be a good idea to cache this somewhere and ensure that mapValues is not an Array of Promise

Reply:
Yes that the point :/ We need to work on that